api/v1alpha1/clusterextension_types.go

```diff
@@ -168,11 +169,11 @@ type ClusterExtensionSpec struct {
 	//
 	// When unspecified, upgrade edges across all channels will be used to identify valid automatic upgrade paths.
 	//
-	// This field follows the DNS subdomain name standard as defined in RFC
-	// 1123, with a deviation in the maximum length being no more than 48
-	// characters. This means that valid values:
-	//   - Contain no more than 48 characters
-	//   - Contain only lowercase alphanumeric characters, '-', or '.'
+	// This field follows the DNS label name standard as defined in [RFC 1123]
+	// with the addition of '.' and '_' characters being allowed.
```

Comment:
I assume `.` and `_` are disallowed as the first or last characters? Seems like that's an important detail to include.